predicators/nsrt_learning/option_learning.py

```diff
@@ -286,12 +288,20 @@ class _KinematicActionConverter(_ActionConverter):
 
     def __init__(self) -> None:
         super().__init__()
+        self._init()
+
+    def _init(self) -> None:
         # Create a new PyBullet connection and robot.
         self._physics_client_id = p.connect(p.DIRECT)
         # Create the robot.
         self._robot = create_single_arm_pybullet_robot(CFG.pybullet_robot,
                                                        self._physics_client_id)
 
+    def __setstate__(self, state: Dict) -> None:
+        # Recreate the object to avoid issues with the PyBullet client.
+        del state  # unused
```

Comment:
This feels like a hack, but I guess it's the best way to get around the problem for now

Reply:
agreed